bot/exts/evergreen/wikipedia.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from typing import List, Optional
+
+from discord import Color, Embed, Message
+from discord.ext import commands
+
+from bot.constants import Wikipedia
+
+log = logging.getLogger(__name__)
+
+SEARCH_API = "https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch={search_term}&format=json"
+WIKIPEDIA_URL = "https://en.wikipedia.org/wiki/{title}"
+
+
+class WikipediaCog(commands.Cog):
+    """Get info from wikipedia."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+    async def search_wikipedia(self, search_term: str) -> Optional[List[str]]:
+        """Search wikipedia and return the first page found."""
+        async with self.http_session.get(SEARCH_API.format(search_term=search_term)) as response:
+            data = await response.json()
+        page = []
+
+        search_results = data["query"]["search"]
+        if len(search_results) == 0:
+            return None
+
+        # we dont like "may refere to" pages.
+        for search_result in search_results:
+            log.info("trying to append titles")
+            if "may refer to" in search_result["snippet"]:
+                pass
+            else:
+                page.append(search_result["title"])
+        log.info("Finished appending titles")
+        return page
+
+    @commands.cooldown(1, 10, commands.BucketType.user)
+    @commands.command(name="wikipedia", aliases=["wiki"])
+    async def wikipedia_search_command(self, ctx: commands.Context, *, search: str) -> None:
+        """Returns list of your search query from wikipedia."""
+        titles_no_underscore: List[str] = []
+        s_desc = ''
+
+        titles = await self.search_wikipedia(search)
+
+        def check(message: Message) -> bool:
+            return message.author.id == ctx.author.id
+
+        if titles is None:
+            await ctx.send("Sorry, we could not find a wikipedia article using that search term")
+            return
+
+        for title in titles:
+            title_for_creating_link = title.replace(" ", "_")  # wikipedia uses "_" as spaces
+            titles_no_underscore.append(title_for_creating_link)
+        log.info("Finished appending titles to titles_no_underscore list")
+
+        async with ctx.typing():
+            for index, title in enumerate(titles, start=1):
+                s_desc += f'`{index}` [{title}]({WIKIPEDIA_URL.format(title=title.replace(" ", "_"))})\n'
+            embed = Embed(colour=Color.blue(), title=f"Wikipedia results for `{search}`", description=s_desc)
+            embed.timestamp = datetime.datetime.utcnow()
+            await ctx.send(embed=embed)
+        embed = Embed(colour=Color.green(), description="Enter number to choose")
+        msg = await ctx.send(embed=embed)
+        chances = 0
+        total_chances = Wikipedia.total_chance
+        l_of_list = len(titles_no_underscore)  # getting length of list
+
+        while chances <= total_chances:
```

Comment:
This could just be a `for _ in range(total_chances )` to remove any chance of infinite loops

Reply:
in constants.py i have stored total chances and its incrementing at every input from user.
User has total of 3 chance if he/she gives 3 wrong input looks breaks, so i dont think is there any chance of infinite loop 
or i am missing anything can u tell me how infinite loop can happen.
Loop breaks if these two thing happens also:
 - If user gives right input 
 - If times up

Reply:
In my opinion `for _ in range(total_chances)` or more precisely `for _ in range(Wikipedia.total_retries)` is a better idiom for this purpose. It's not that probable that somehow, somewhere, something is going to error out and cause an infinite loop but then again, using common idioms helps better with readability.

Reply:
ok